From 5c63344444a5fb7807cb457dbc60f3bfbd98d4c2 Mon Sep 17 00:00:00 2001
From: ghost <andreas.monzner@dream-property.net>
Date: Fri, 15 Mar 2013 11:22:43 +0100
Subject: [PATCH 3/3] mytest.py: since we have removed
 ePixmap::setPixmap(gPixmap *) it is not allowed to set
 setPixmap(None) in python.. so we add a simple wrapper
 to re-allow this..

---
 mytest.py |   11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/usr/lib/enigma2/python/mytest.py b/usr/lib/enigma2/python/mytest.py
index 39d544d..0cf4679 100644
--- a/usr/lib/enigma2/python/mytest.py
+++ b/usr/lib/enigma2/python/mytest.py
@@ -11,6 +11,17 @@ def gPixmapPtr_deref(self):
 
 enigma.gPixmapPtr.__deref__ = gPixmapPtr_deref
 
+ePixmap_setPixmap_org = enigma.ePixmap.setPixmap
+
+def ePixmap_setPixmap(self, pixmap):
+	if pixmap is None:
+		pm = enigma.gPixmapPtr()
+		ePixmap_setPixmap_org(self, pm)
+	else:
+		ePixmap_setPixmap_org(self, pixmap)
+
+enigma.ePixmap.setPixmap = ePixmap_setPixmap
+
 from Tools.Profile import profile, profile_final
 
 profile("PYTHON_START")
-- 
1.7.10.4

